docs/data/toolpad/core/components/sign-in-page/sign-in-page.md

```diff
@@ -0,0 +1,109 @@
+---
+productId: toolpad-core
+title: Sign In Page
+components: SignInPage, Account, NotificationsProvider
+---
+
+# Sign In Page
+
+<p class="description">A component that renders a functional authentication page for your application.</p>
+
+The `SignInPage` component is a quick way to generate a ready-to-use authentication page with multiple OAuth providers, or a credentials form.
+
+## OAuth
+
+The `SignInPage` component can be set up with an OAuth provider by passing in a list of providers in the `providers` prop, along with a `signIn` function that accepts the `provider` as a parameter.
+
+{{"demo": "OAuthSignInPage.js", "iframe": true}}
+
+## Credentials
+
+:::warning
+It is recommended to use the OAuth provider for more robust maintenance, support, and security.
+:::
+
+To render a username password form, pass in a provider with `credentials` as the `id` property. The `signIn` function accepts a `formData` parameter in this case.
+
+{{"demo": "CredentialsSignInPage.js", "iframe": true, "height": 500}}
+
+### Alerts
+
+The `signIn` prop takes a function which can either return `void` or a `Promise` which resolves with an `AuthResponse` object of the form:
+
+```ts
+interface AuthResponse {
+  error?: string;
+  type?: string;
+}
+```
+
+This renders an alert with the `error` string as the message.
+
+{{"demo": "NotificationsSignInPageError.js", "iframe": true, "height": 600}}
+
+## Usage with authentication libraries
+
+### Auth.js
+
+#### Next.js App Directory and GitHub
+
+The component is composable with any authentication library you might want to use. The following is a functional `SignInPage` with [auth.js](https://authjs.dev/) using GitHub, Next.js App router and server actions.
+
+:::warning
+The following demo does not initiate an actual GitHub authentication flow, since doing that from within an `iframe` is not permitted. Run the [Next.js app directory](https://github.com/mui/mui-toolpad/tree/master/examples/core-auth-nextjs/) example to test this functionality.
+:::
+
+{{"demo": "AuthJsSignInApp.js", "iframe": true, "height": 300 }}
+
+#### Setting up
```

Comment:
I know this goes against my first comment on this, but I think maybe we can skip these detailed setup instructions and instead just link to the Next auth documentation? Or maybe it's actually nice to provide these detailed instructions as an example, but we should make it clearer that users should just follow the Next Auth docs as they're implementing the library directly.
Or maybe these detailed instructions should only be included with the examples themselves and not the component documentation.

Reply:
I can definitely cut down on the instructions, give the link to the Next Auth docs more priority and keep the detailed instructions only in the example.